android/src/main/java/com/wenkesj/voice/VoiceModule.java

```diff
@@ -159,31 +175,52 @@ private void sendEvent(String eventName, @Nullable WritableMap params) {
 
   @Override
   public void onBeginningOfSpeech() {
-    WritableMap event = Arguments.createMap();
-    event.putBoolean("error", false);
-    sendEvent("onSpeechStart", event);
+    // The beginning of automatic detected speech
+    // WritableMap event = Arguments.createMap();
+    // event.putBoolean("error", false);
+    // sendEvent("onSpeechStart", event);
+    Log.d("ASR", "onBeginningOfSpeech()");
   }
 
   @Override
   public void onBufferReceived(byte[] buffer) {
     WritableMap event = Arguments.createMap();
     event.putBoolean("error", false);
     sendEvent("onSpeechRecognized", event);
+    Log.d("ASR", "onBufferReceived()");
   }
 
   @Override
   public void onEndOfSpeech() {
-    WritableMap event = Arguments.createMap();
-    event.putBoolean("error", false);
-    sendEvent("onSpeechEnd", event);
+    if (isRecognizing) {
+      final VoiceModule self = this;
+      Handler mainHandler = new Handler(this.reactContext.getMainLooper());
+      mainHandler.post(new Runnable() {
+        @Override
+        public void run() {
+          speech.cancel();
```

Comment:
So, the end of speech event is captured during the recognition. We re-start the startListening() right?

Reply:
Yes. I have seen that after `onEndOfSpeech`, speech recognizer does not recognize anymore. Thus, I restart `startListening()`.

Reply:
Got it.